crates/net/src/sim2h_worker.rs

```diff
@@ -209,58 +232,65 @@ impl Sim2hWorker {
-            did_something = true;
-            let message = self.outgoing_message_buffer.get(0).unwrap();
-            debug!("WireMessage: preparing to send {:?}", message);
-            let payload: String = message.clone().into();
-            let signature = self
-                .conductor_api
-                .execute(payload.clone(), CryptoMethod::Sign)
-                .unwrap_or_else(|e| {
-                    panic!(
-                        "Couldn't sign wire message in sim2h worker: payload={}, error={:?}",
-                        payload, e
-                    )
-                });
-            let signed_wire_message = SignedWireMessage::new(
-                message.clone(),
-                Provenance::new(self.agent_id.clone(), signature.into()),
+        }
+        let message = &buffered_message.wire_message;
+        debug!("WireMessage: preparing to send {:?}", message);
+        let payload: String = message.clone().into();
+        let signature = self
+            .conductor_api
+            .execute(payload.clone(), CryptoMethod::Sign)
+            .unwrap_or_else(|e| {
+                panic!(
+                    "Couldn't sign wire message in sim2h worker: payload={}, error={:?}",
+                    payload, e
+                )
+            });
+        let payload: Opaque = payload.into();
+        let signed_wire_message = SignedWireMessage::new(
+            payload.clone(),
+            Provenance::new(self.agent_id.clone(), signature.into()),
+        );
+        let to_send: Opaque = signed_wire_message.into();
+
+        // safe to unwrap because we check connection_ready() above
+        if let Err(e) = self
+            .connection
+            .as_mut()
+            .unwrap()
+            .write(to_send.to_vec().into())
+        {
+            error!(
+                "TransportError trying to send message to sim2h server: {:?}",
+                e
             );
-            let to_send: Opaque = signed_wire_message.into();
-            // safe to unwrap because we check connection_ready() above
-            if let Err(e) = self
-                .connection
-                .as_mut()
-                .unwrap()
-                .write(to_send.to_vec().into())
-            {
-                error!(
-                    "TransportError trying to send message to sim2h server: {:?}",
-                    e
-                );
-                self.connection = None;
-                self.check_reconnect();
-                return did_something;
-            }
-            debug!("WireMessage: dequeuing sent message {:?}", message);
-            // if we made it here, we successfully sent the first message
-            // we can remove it from the outgoing buffer queue
-            self.outgoing_message_buffer.remove(0);
+            self.connection = None;
+            self.check_reconnect();
+            return true;
         }
+        #[allow(deprecated)]
+        let mut hasher = SipHasher::new();
```

Comment:
vote for going with the siphasher crate https://crates.io/crates/siphasher to avoid the deprecation

alternately https://crates.io/crates/twox-hash because it is faster for larger byte-count messages

Reply:
xxhash starts being faster at ~16 or 32 bytes - the tracer context by itself is 37 bytes.